src/engine/enginebufferscalelinear.cpp

```diff
@@ -232,34 +229,34 @@ CSAMPLE* EngineBufferScaleLinear::do_scale(CSAMPLE* buf,
 
         // update our class members so next time we need to scale it's ok. we do
         // blow away the fractional sample position here
-        buffer_int_size = 0; // force buffer read
-        m_dNextSampleIndex = 0;
+        m_bufferIntSize = 0; // force buffer read
+        m_dNextFrame = 0;
         m_fPrevSample[0] = buf[read_samples-2];
         m_fPrevSample[1] = buf[read_samples-1];
         return buf;
     }
 
-    // Simulate the loop to estimate how many samples we need
-    double samples = 0;
+    // Simulate the loop to estimate how many frames we need
+    double frames = 0;
 
+    // We're calculating frames = 2 samples, so divide remaining buffer by 2;
     for (int j = 0; j < iRateLerpLength; j += 2) {
-        samples += fabs((rate_add_diff * static_cast<float>(j)) /
+        frames += fabs((rate_add_diff * static_cast<float>(j)) /
                         static_cast<float>(iRateLerpLength) + rate_add_old);
     }
 
-    // We're calculating mono samples, so divide remaining buffer by 2;
-    long unscaled_samples_needed = floor(samples);
+    int unscaled_frames_needed = floor(frames);
 
     // If the current position fraction plus the future position fraction
     // loops over 1.0, we need to round up
-    if (m_dNextSampleIndex - floor(m_dNextSampleIndex) +
-        samples - floor(samples) > 1.0) {
-        unscaled_samples_needed++;
+    if (m_dNextFrame - floor(m_dNextFrame) +
+            frames - floor(frames) > 1.0) {
+        unscaled_frames_needed++;
     }
 
     // Multiply by 2 because it is predicting mono rates, while we want a stereo
     // number of samples.
-    unscaled_samples_needed *= 2;
+    int unscaled_samples_needed = unscaled_frames_needed * 2;
 
     // 0 is never the right answer
     unscaled_samples_needed = math_max<long>(2, unscaled_samples_needed);
```

Comment:
ok to combine this line with the one above, even though I'm sure the compiler has figured it out.
int unscaled_samples_needed = math_max<long>(2, unscaled_frames_needed * 2);